test/framework/workload.go

```diff
@@ -63,6 +67,42 @@ func (w *WorkloadCluster) WaitForKubeconfig() {
 	if err != nil {
 		w.T.Fatalf("Failed waiting for cluster kubeconfig: %s", err)
 	}
+	w.initClusterStateConfig(ctx)
+}
+
+// ValidateClusterDelete verifies the cluster has been deleted.
+func (w *WorkloadCluster) ValidateClusterDelete() {
+	ctx := context.Background()
+	w.T.Logf("Validating cluster deletion %s", w.ClusterName)
+	if w.clusterStateConfig == nil {
+		return
+	}
+	clusterStateValidator := newClusterStateValidator(w.clusterStateConfig)
+	WithValidationsForClusterDoesNotExist(clusterStateValidator)
+	if err := clusterStateValidator.Validate(ctx); err != nil {
+		w.T.Fatalf("failed to validate cluster deletion %v", err)
+	}
+	w.clusterStateConfig = nil
+}
+
+func (w *WorkloadCluster) initClusterStateConfig(ctx context.Context) {
```

Comment:
same here, I would avoid this and just have a method that returns the clusterStateConfig